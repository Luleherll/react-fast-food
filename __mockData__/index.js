const user = {
  user:'token_here',
  Admin: true
};

const orders = [{
  amount: 6000,
  comment: 'Include cabbages please',
  created_at: 'Saturday, 27. October 2018 09:43AM',
  ended_at: 'Saturday, 27. October 2018 09:44AM',
  food_id: 32,
  img1: 'https://res.cloudinary.com/lule/image/upload/v1540621281/rolex2_wuewwf.jpg',
  location: 'Bukoto',
  name: 'UG rolex',
  order_id: 19,
  quantity: 2,
  status: 'Declined',
  user_id: 1
}];
const food = [
  {
    food_id:29,
    img1:'https://res.cloudinary.com/lule/image/upload/v1540617122/Pitza_oasfp5.jpg',
    img2:'https://res.cloudinary.com/lule/image/upload/v1540617144/pitza2_e750os.jpg',
    img3:'https://res.cloudinary.com/lule/image/upload/v1540617160/pitza3_a9hq5q.jpg',
    name:'Pizza',
    price:20000,
    status:'Available',
    tags:'meal'
  }
];

const initialState = {
  food: {
    foodItems: {},
    userPendingOrders: {},
    userOrderHistory: {},
  },
  orders: {
    newOrders: orders,
    allPendingOrders: [{...orders[0], status: 'Pending'}],
    allArchivedOrders: [{...orders[0], status: 'Declined'}],
  }
};

export {user, orders, food, initialState};