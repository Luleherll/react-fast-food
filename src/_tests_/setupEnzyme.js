import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';


Enzyme.configure({ adapter: new Adapter() });

// Object.defineProperty(global.document, 'getElementById', {
//   value: jest.fn(() => ({ innerHTML: ['none'] })),
//   writable: true,
// });

// Object.defineProperty(global.document, 'execCommand', {
//   value: jest.fn(),
//   writable: true,
// });

// Object.defineProperty(global, 'prompt', { value: jest.fn(() => ['none']) });

// fontawesome.library.add(solid, regular);

// global.fontawesome = fontawesome;

export { shallow, mount, render };