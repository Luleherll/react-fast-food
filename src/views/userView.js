import React from 'react';
import {connect} from 'react-redux';
import * as AllActions from '../redux/Thunks';
import { bindActionCreators } from 'redux';
import Header from '../components/common/Header';
import AdminDashboard from '../components/admin';
import UserDashboard from '../components/customer';
import M from 'materialize-css';


export class UserView extends React.Component{
    state = {
      nav : {
        navItem1: {name: 'menu', action: ()=>this.displayPage('foodItems', 'foodItems')},
        navItem2: {name: 'myOrders'},
        navItem3: {name: 'history'}
      },
      adminProp: this.props.newOrders || [],
      foodItems: this.props.foodItems || []
    }

    componentDidMount(){
      document.addEventListener('click', () => {
        let elems = document.querySelectorAll('.sidenav');
        M.Sidenav.init(elems);
      });
      const Admin = localStorage.getItem('Admin');
      if (Admin==='true') {
        this.setState({
          nav:{
            navItem1: {name: 'Orders', action: ()=>this.displayPage('adminProp', 'newOrders')},
            navItem2: {name: 'pending', action: ()=>this.displayPage('adminProp', 'allPendingOrders')},
            navItem3: {name: 'archived', action: ()=>this.displayPage('adminProp', 'allArchivedOrders')}
          }
        });
      }
    }

    displayPage = (prop, page) => this.setState({ [prop]: this.props[page] || [] });
    logOut = () =>{
      localStorage.clear();
      this.props.history.push('/');
    }
    getRoute = (pathname, adminProps, userProps) =>{
      switch (pathname) {
        case 'admin':
          return(
            <AdminDashboard {...adminProps}/>
          );

        case 'user':
          return(<UserDashboard foodItems={userProps}/>);
      
        default:
          break;
      }
    }
    render(){
      const {adminProp, nav, foodItems} = this.state;
      const adminProps = {adminProp};
      const route = this.props.match.params.route;
      return(
        <div id='dash'>
          <Header logout={this.logOut} nav={nav}/>
          {this.getRoute(route, adminProps, foodItems)}
        </div>
      );
    }
}

const mapStateToProps = ({food, orders}) =>({
  foodItems: food.foodItems,
  userPendingOrders: food.userPendingOrders,
  userOrderHistory: food.userOrderHistory,
  newOrders: orders.newOrders,
  allPendingOrders: orders.allPendingOrders,
  allArchivedOrders: orders.allArchivedOrders,
});
function mapDispatchToProps(dispatch) {
  return {actions: bindActionCreators(AllActions, dispatch)};
}

export default connect(mapStateToProps, mapDispatchToProps)(UserView);