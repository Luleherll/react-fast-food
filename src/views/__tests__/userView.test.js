import React from 'react';
import {mount} from '../../_tests_/setupEnzyme';
import UserViewTest, {UserView} from '../userView';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import {initialState} from '../../../__mockData__';
import userView from '../userView';


const headerOrderButtons = ['#newOrder','#pendingOrder', '#doneOrder'];
let store;
const mockStore = configureStore([thunk]);
const props = {
  actions: {
    getDataThunk: jest.fn(),
    postDataThunk: jest.fn(),
  },
  match: {params: {route: ''}}
};


jest.useFakeTimers();
describe('mainView', () => {
  let wrapper;

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(<Provider store={store}><UserViewTest {...props}/></Provider>);
    localStorage.setItem('Admin', 'true');
  });

  it('should renders correctly', () => {
    expect(wrapper.find('.collapsible')).toBeDefined();
  });
  
  headerOrderButtons.forEach(button =>(
    it('displayPage is called when order button is clicked', () => {
      wrapper = mount(<UserView {...props} />);
      jest.spyOn(wrapper.instance(), 'displayPage');
      wrapper.find(button).simulate('click');
    })
  ));
  it('navbar items change', () => {
    localStorage.setItem('Admin', 'false');
    expect(wrapper).toMatchSnapshot();
  });
});