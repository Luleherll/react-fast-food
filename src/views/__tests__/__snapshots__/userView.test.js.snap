// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainView navbar items change 1`] = `ReactWrapper {}`;
